src/mnesia_rocksdb.erl

```diff
@@ -1555,6 +1571,55 @@ keypat_pfx({HeadPat,_Gs,_}, KeyPos) when is_tuple(HeadPat) ->
 keypat_pfx(_, _) ->
     <<>>.
 
+
+%% ----------------------------------------------------------------------------
+%% Db wrappers
+%% ----------------------------------------------------------------------------
+return(_Res, badarg) ->
+    badarg;
+return(Res, _) ->
+    Res.
+
+db_put(Ref, K, V, Opts, St) ->
+    write_result(?rocksdb:put(Ref, K, V, Opts), put, [Ref, K, V, Opts], St).
+
+db_write(Ref, List, Opts, St) ->
+    write_result(?rocksdb:write(Ref, List, Opts, St), write, [Ref, List, Opts], St).
+
+db_delete(Ref, K, Opts, St) ->
+    write_result(?rocksdb:delete(Ref, K, Opts), delete, [Ref, K, Opts], St).
+
+write_result(ok, _, _, _) ->
+    ok;
+write_result(Res, Op, Args, #st{on_write_error = Rpt}) ->
+    RptOp = rpt_op(Rpt),
+    mnesia_lib:RptOp("FAILED rocksdb:~p(" ++ rpt_fmt(Args) ++ ") -> ~p~n",
+                   [Op | Args] ++ [Res]),
+    if Rpt == fatal; Rpt == error ->
+            badarg;
```

Comment:
A generic `badarg` to cover all error cases will make handling errors in clients quite hard. Is there no way to pass the error further up the call chain?

Reply:
The only places where the callback module can trigger a client-side error are `validate_key/6` and `validate_record/6`. The actual update operations happen after the commit point, so the alternatives are basically:
* fail silently
* log/report, then fail silently
* log/report, then crash - `badarg` - causing termination of the calling process (internal mnesia error)
* dump core

Reply:
Ok, so in the client we can only retry. If that fails too we have to bail out entirely. Or we build a lager tracer which inspects log messages from Mnesia and can be queried given a transaction identifier.